dali/python/nvidia/dali/pool.py

```diff
@@ -0,0 +1,327 @@
+# Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import socket
+import pickle
+import threading
+import multiprocessing
+from collections import OrderedDict
+from nvidia.dali import backend as _b
+from nvidia.dali.worker import worker
+from nvidia.dali.shared_batch import deserialize_batch, import_numpy
+from nvidia.dali import shared_mem
+
+
+class MemChunk:
+
+    def __init__(self, shm_chunk, capacity):
+        self.shm_chunk = shm_chunk
+        self.capacity = capacity
+
+
+class SharedBatchesConsumer:
+    """
+Counterpart of worker.py:SharedBatchesDispatcher. Can receive and deserialize batch from the worker,
+keeps track of already received memory chunks and opens new chunks or resizes exiting ones if necessary.
+"""
+
+    def __init__(self):
+        import_numpy()
```

Comment:
I wonder if we can silently introduce such dependency. On the other hand if the user wants to use ExternalSource he usually have numpy installed already.
Maybe we should warn that this functionality requires numpy is the import fails?

Reply:
Added exception with more specific message to imprt_numpy function (shared_batch.py:22)